conftest.py

```diff
@@ -122,26 +127,11 @@ def _get_s3_provider(request):
     return _get_storage_provider(request, S3)
 
 
-@pytest.fixture
-def memory_storage(request):
-    if not _is_opt_true(request, MEMORY_OPT):
-        return _get_memory_provider(request)
-
-
-@pytest.fixture
-def local_storage(request):
-    if _is_opt_true(request, LOCAL_OPT):
-        return _get_local_provider(request)
-
-
-@pytest.fixture
-def s3_storage(request):
-    if _is_opt_true(request, S3_OPT):
-        return _get_s3_provider(request)
+def _get_dataset(provider: StorageProvider):
+    return Dataset(path=str(uuid1()), provider=provider)
```

Comment:
is there a reason for having a uuid path here? the `provider` already manages the paths exactly how we want them. i feel like if you use a uuid here it may just make things more confusing

Reply:
i think you probably have it here because path can't be empty, but i think that if you provide a provider you should be able to leave path empty.